instrumentation/google.golang.org/grpc/otelgrpc/interceptor.go

```diff
@@ -366,31 +366,31 @@ func UnaryServerInterceptor(opts ...Option) grpc.UnaryServerInterceptor {
 			messageReceived.Event(ctx, 1, req)
 		}
 
-		var statusCode grpc_codes.Code
-		defer func(t time.Time) {
-			elapsedTime := time.Since(t) / time.Millisecond
-			attr = append(attr, semconv.RPCGRPCStatusCodeKey.Int64(int64(statusCode)))
-			o := metric.WithAttributes(attr...)
-			cfg.rpcServerDuration.Record(ctx, int64(elapsedTime), o)
-		}(time.Now())
+		before := time.Now()
+		var grpcStatusCode grpc_codes.Code
 
 		resp, err := handler(ctx, req)
 		if err != nil {
 			s, _ := status.FromError(err)
+			grpcStatusCode = s.Code()
 			statusCode, msg := serverStatus(s)
 			span.SetStatus(statusCode, msg)
 			span.SetAttributes(statusCodeAttr(s.Code()))
```

Comment:
This is going to set the status code. Why duplicate this?

Reply:
So you mean to change it to `span.SetAttributes(statusCodeAttr(grpcStatusCode))`?

I can do that.

If you mean other things, please explain more :)

Reply:
Yeah, that change sounds good to me. It should be unified outside of this conditional though (i.e. on line 391 or near by).

Reply:
I see. Noticing line 390 to 392 are for that `rpc.server.duration` metric only. They have nothing to do with tracing or the `span` here.

Reply:
I made a simple change here to use `grpcStatusCode` instead `s.Code()` or `grpc_codes.OK` below.

Otherwise, one could also put this `s, _ := status.FromError(err)`
outside on the top of the `if` block, as `status.FromError(err)` returns an effective `OK` status.

But as there are multiple other places in this file that are written with a similar pattern, I will just keep the changes in this PR relatively small and not change too much here.

Hope that makes sense :)

Reply:
If we are unifying the code into a var I am in favor of unifying its use outside of the conditional. Please update with the simplifications you mentioned.

Reply:
Oh thanks, do you mean I do it for the whole file or just this part of code?

Reply:
I made a small update on the code part I changed cff17fa, not touching on other similar places in this file.